### DISCUSSION TITLE
How can I custom terminal style?

### DISCUSSION THREAD

**Discussion by fish-404** [codesandbox/codesandbox-client#5987] [Q&A] [answered]
How can I custom terminal style by myself like editor?
The default terminal font space setting seems too large, hard to read.

**Comment by smhutch** [comment, marked as answer]
Hi @fish-404, we don't currently support custom terminal styles

**Reply by fish-404** [reply]
Thank you. Is possible open terminal styles to users? Like using a configure file?

**Comment by lbogdan** [comment]
Hey @fish-404 ,

What OS and browser are you using? Can you please attach a screenshow of how the terminal renders for you? Thanks!

**Reply by fish-404** [reply]
OS: Windows 10 Enterprise 20H2 19042.1165
Browser: Google Chrome 93.0.4577.63 32bit
The space bettween letters seems a little large. 
FYI. Thank you~

![image](https://user-images.githubusercontent.com/29678177/132458642-6977c493-9cb3-4515-a220-dd4f8047544a.png)

**Reply by JamesACS** [reply]
Hey @fish-404 - Unfortunately we haven't been able to replicate this - are you still experiencing the issue?

**Reply by fish-404** [reply]
Hi, I still experience this issue. In fact, I always see this issuses like the screenshot. 
I ​searched some screenshot in the web. Seems other users termianls are normal. 
Do you have any ideas about network or some computer resitriction could cause this? I experience this issue in my working computer (I add screenshots for Firefox and Edge below). I will check if my personal PC will meet this.  Thank you. 

> Firefox 91.0.2 (64-bit)

![image](https://user-images.githubusercontent.com/29678177/136876133-7390f992-2ab0-400f-9b04-a69cd237cbe8.png)

>Microsoft Edge 94.0.992.38 (Official build) (64-bit)

![image](https://user-images.githubusercontent.com/29678177/136876299-ed114f4c-136e-4071-9125-4c7310ee7ba1.png)

**Reply by JamesACS** [reply]
Definitely interesting behaviour - But I haven't been able to replicate this on Edge, Firefox or Chrome so it might be a local setting - I'm afraid I wouldn't be able to diagnose that though, especially if it's a work computer. 

Could you let us know whether it appears on your home device as well?

**Reply by fish-404** [reply]
Hi, I still meet this issue in my personal computer. Confused. I have no idea about this. 
But my editor is normal. I paste screen shots below. Do you need more information?

![image](https://user-images.githubusercontent.com/29678177/136996887-a4abc658-884a-4c8f-8182-c8cddaab09d6.png)


>OS: Windows 10 Home Chinese Version 1909 18363.1556

>Firefox: 83.0 (64-bit)

![image](https://user-images.githubusercontent.com/29678177/136996804-1ff80d35-197b-4525-baf4-137876c80f74.png)

>Edge: 94.0.992.38 (64-bit)

![image](https://user-images.githubusercontent.com/29678177/136996985-e4b27cba-127b-449b-8dd8-c0951587c4fc.png)

>Chrome: 94.0.992.38 (64-bit)

![image](https://user-images.githubusercontent.com/29678177/136997284-c3056d30-7522-4ec6-88c3-6b18edad1859.png)

**Reply by JamesACS** [reply]
@fish-404 If this is only the one Computer it's likely to be a local issue. Unfortunately without knowing further reproduction steps we haven't been able to figure this out internally.

There is a Chrome help article here which gives a number of suggestions: https://support.google.com/chrome/answer/95290?hl=en-GB

**Reply by fish-404** [reply]
Thanks for you advise. But if it's a local issue, like the Chrome help article pointed, I think other text will show the similar style. 
My CodeSanbox's edior works perfect, only teminal looks strange. 
By the way, do you think if network will affect the fonts style? Chinese network limitation will make the network a little different from yours.
May be I would ask some friends to check if they will see the strange font style.

**Reply by JamesACS** [reply]
@fish-404 if you do, please report back. 

Unfortunately we can't test for any network configuration issues but broadly speaking how your network is configured shouldn't effect how text displays on a website.

**Reply by fish-404** [reply]
@JamesACS Hi, I find the font style is normal now! I change nothing. 
![image](https://user-images.githubusercontent.com/29678177/150891517-b3f8132e-ab02-481f-acdf-21eff6eff232.png)

**Comment by JamesACS** [comment]
That's great news @fish-404 - If it occurs again please let us know

**Reply by fish-404** [reply]
@JamesACS  Hi, I find something more interesting.  Now I can see the normal style in my company computer, but still strange in my personal computer.  I'm not sure if my company network changes. But my company network may a litte different with our personal network (I mean Chinese special network limit).

And I bought a new computer recently, I set nothing special now. When I use Edge in this computer, style also looks strange. 
I capture a console screen below, see if it can help. Maybe some resource can't find because of  network?

Visit CodeSandbox with Edge
![Visit CodeSandbox with Edge](https://user-images.githubusercontent.com/29678177/156389261-f3aa00ab-9c33-4ad5-94fd-8f1c9d841972.png)

Base on network suspection, I take an expirement in my old computer Firefox. 
* I visit CodeSandbox with proxy, then I can see the normal style
* When I visit CodeSandbox with no proxy (doesn't open a new private window, just refresh), I can see the normal style
* I open a new private window and visit CodeSandbox again, it return to strange style
* I visit CodeSandbox with proxy again in this private window, it turn to normal !

Visit CodeSandbox with proxy
![Visit CodeSandbox with proxy](https://user-images.githubusercontent.com/29678177/156392797-f479d14b-4be2-42eb-b6f7-f5092fbb8cb9.jpg)

Visit CodeSandbox no proxy
![Visit CodeSandbox no proxy](https://user-images.githubusercontent.com/29678177/156392864-b1d384ae-974b-4e0c-9806-4e123af46d81.jpg)

**Reply by JamesACS** [reply]
Thanks for sending that through @fish-404 - Based on what you've sent this would definitely be a network issue on your end. Have you tried clearing your DNS cache?

**Reply by fish-404** [reply]
@JamesACS  Hi, I try to clear DNS cache, but it doesn't work. I'm curious about why the style will looks different between with and without proxy, and styles difference only shown in Terminal widget.

**Reply by JamesACS** [reply]
Thanks @fish-404 - I'm afraid I wouldn't know a solution in this case. It could be due to something blocked by your ISP but as it's only happening in one browser on one device, it's likely there is an extension causing a conflict or something on your network.

**Reply by fish-404** [reply]
Thanks for following~ Will info you if I find something new may help reproduce this situation. 
> It could be due to something blocked by your ISP 

I think this is the reason, but it's not "only happen in one browser on one device". I have reproduce this situation more than one device even I haven't install extensions.

**Reply by JamesACS** [reply]
Thanks @fish-404
